_posts/2022-08-14-aug-update.markdown

```diff
@@ -0,0 +1,171 @@
+
+As a blog post without images would be boring, here are some screenshots, first off is Managarm running `python`.
+![python](/assets/2022-aug-update/python.png)
+After that, we have `xclock`.
+![xclock](/assets/2022-aug-update/xclock.png)
+And finally we have `exa` running.
+![exa](/assets/2022-aug-update/exa.png)
+
+### The road to X11
+The road to X11 was quite a bumpy one, with several issues that required digging deep into the X11 codebase. In the end, the biggest issues were a nasty epoll bug and the usage of abstract UNIX sockets, that were not implemented yet. With that fixed (and a small amount of stubbing of shared memory functions in mlibc) we were able to run the `gtk-demo` demo program successfully, paving the way for various other X based programs.
+![gtk2](/assets/2022-aug-update/gtk2.png)
+
+Outside of XWayland, work is ongoing to also run the classic X.Org server, while using its own DRM-based mode setting (instead of Weston's).
+
+### QEMU
+Most of the pieces necessary for QEMU have already been in place, with the exception of `sigaltstack` and partial `munmap`/`mmap`/`mprotect` support. With both of these missing features implemented, we can run QEMU on Managarm, bringing us one step closer to being self-hosting.
+
+![qemu](/assets/2022-aug-update/qemu.png)
+
+### DOOM
+Until recently, we did not have any DOOM port, mainly because we could not decide on which source port to use. We eventually decided upon dsda-doom, giving us a modern, yet vanilla DOOM experience, with extra speedrunning features as a bonus.
+
+![doom](/assets/2022-aug-update/doom.png)
+
+## What do we want to achieve in 2022?
+
+**Finish porting xbps.** As mentioned above, considerable work went into porting the xbps package manager. While the general infrastructure, both inside Managarm and outside in terms of an repository, are set up, some more work is required to actually get `xbps` to function properly. We aim to implement the missing functionality soon.
+
+**Polish the port collection.** Currently, we have a lot of ports that work at least partially, but some ports use pretty ugly hacks to get to that state. We should strive to get the quality of these ports up by improving functionality within mlibc and by removing hacks. This also includes adding better tests to check for correctness. We also plan to start upstreaming support patches such that we can remove some patches from the collection.
+
+**Complete TTY subsystem.** We currently lack or incorrectly implement many TTY subsystem features (sessions and signals, process groups, et cetera), which are quite necessary for many kinds of programs as well as day-to-day life using the system. This goal is also accompanied by finishing up Unix process credentials and signals.
+
```

Comment:
@Dennisbonke: add something about browsers here?

Reply:
Nope. I want no major content changes at all. Ideally I'd not talk about things after January 2022 as to keep content for the end of 2022 post but what can you do about it. I'll write some browser stuff for the next post.